x-pack/plugins/cases/public/containers/use_get_categories.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+import { useToasts } from '../common/lib/kibana';
+import { useCasesContext } from '../components/cases_context/use_cases_context';
+import type { ServerError } from '../types';
+import { getCategories } from './api';
+import { casesQueriesKeys } from './constants';
+import * as i18n from './translations';
+
+export const useGetCategories = () => {
+  const toasts = useToasts();
+  const { owner } = useCasesContext();
+  return useQuery(
+    casesQueriesKeys.categories(),
+    () => {
+      const abortCtrl = new AbortController();
+      return getCategories(abortCtrl.signal, owner);
+    },
+    {
```

Comment:
I think we should add `staleTime: 60 * 1000, // one minute` or more as categories do not change that often. We need to verify though that whenever we add/update a category we invalidate the query correctly.